src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureDeserializerJavascriptSerializerWithSimpleTypeResolver.cs

```diff
@@ -0,0 +1,250 @@
+                    {
+                        return;
+                    }
+
+                    // If JavaScriptSerializer is initialized with a SimpleTypeResolver, then that instance is flagged.
+                    ConstructorMapper constructorMapper = new ConstructorMapper(
+                        (IMethodSymbol constructorMethod, IReadOnlyList<PointsToAbstractValue> argumentPointsToAbstractValues) =>
+                        {
+                            PropertySetAbstractValueKind kind;
+                            if (constructorMethod.Parameters.Length == 0)
+                            {
+                                kind = PropertySetAbstractValueKind.Unflagged;
+                            }
+                            else if (constructorMethod.Parameters.Length == 1
+                                && javaScriptTypeResolverSymbol.Equals(constructorMethod.Parameters[0].Type))
+                            {
+                                PointsToAbstractValue pointsTo = argumentPointsToAbstractValues[0];
+                                switch (pointsTo.Kind)
+                                {
+                                    case PointsToAbstractValueKind.Invalid:
+                                        kind = PropertySetAbstractValueKind.Unflagged;
+                                        break;
+
+                                    case PointsToAbstractValueKind.KnownLocations:
+                                        if (pointsTo.Locations.Any(l => !l.IsNull && simpleTypeResolverSymbol.Equals(l.LocationTypeOpt)))
+                                        {
+                                            kind = PropertySetAbstractValueKind.Flagged;
+                                        }
+                                        else if (pointsTo.Locations.Any(l =>
+                                                    !l.IsNull
+                                                    && javaScriptTypeResolverSymbol.Equals(l.LocationTypeOpt)
+                                                    && (l.CreationOpt == null || l.CreationOpt.Kind != OperationKind.ObjectCreation)))
+                                        {
+                                            // Points to a JavaScriptTypeResolver, but we don't know if the instance is a SimpleTypeResolver.
+                                            kind = PropertySetAbstractValueKind.MaybeFlagged;
+                                        }
+                                        else
+                                        {
+                                            kind = PropertySetAbstractValueKind.Unflagged;
+                                        }
+
+                                        break;
+
+                                    default:
+                                        Debug.Fail($"Unhandled PointsToAbstractValueKind {pointsTo.Kind}");
```

Comment:
So you don't expect other kinds, such as `PointsToAbstractValueKind.Unknown`? Will that not be common in presence of conservative analysis where PointsToAnalysis is unable to figure out the points to value and conservatively assumes unknown.

Reply:
Hrm, I need more test cases for this.